metpy/calc/tools.py

```diff
@@ -109,4 +115,13 @@ def find_intersections(x, a, b):
     # causes weirder unit behavior and seems a little less good numerically.
     intersect_y = ((intersect_x - x0) / (x1 - x0)) * (a1 - a0) + a0
 
-    return intersect_x, intersect_y
+    # Make a mask based on the direction of sign change desired
+    if direction == 'increasing':
+        mask = sign_change > 0
+    elif direction == 'decreasing':
+        mask = sign_change < 0
+    elif direction == 'all':
+        mask = np.ones_like(intersect_x, dtype=np.bool)
```

Comment:
I get the idea that returning from one spot is nice, but there's no guarantee that this extra array allocation is cheap, or even feasible. Just do the direct return in the case of 'all'.